### PR TITLE
fix: Apply max width to confirm modal so that children's width will not increase dramatically

### LINKED ISSUE #51912: Carousel in Modal without icon do not display it
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/p/sandbox/antd-reproduction-template-forked-fph2gd?file=%2Findex.js%3A16%2C1)

### Steps to reproduce

A Carousel set inside a modal used in a contextHolder.
The modal icon is set to null
The carousel have at lest 2 items
Click on the button `open Modal not working` from reproducible link 

### What is expected?

The carousel is displayed and autoplay works

### What is actually happening?

The carousel is then displayed for some seconds and after it disappear

| Environment | Info |
| --- | --- |
| antd | 5.22.3 |
| React | 18 |
| System | Windows |
| Browser | Firefox |

### PR DESCRIPTION
**Pull request by quan060798**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE_CN.md?plain=1)

### 🤔 This is a ...

- [ ] 🆕 New feature
- [x] 🐞 Bug fix
- [ ] 📝 Site / documentation improvement
- [ ] 📽️ Demo improvement
- [ ] 💄 Component style improvement
- [ ] 🤖 TypeScript definition improvement
- [ ] 📦 Bundle size optimization
- [ ] ⚡️ Performance optimization
- [ ] ⭐️ Feature enhancement
- [ ] 🌐 Internationalization
- [ ] 🛠 Refactoring
- [ ] 🎨 Code style optimization
- [ ] ✅ Test Case
- [ ] 🔀 Branch merge
- [ ] ⏩ Workflow
- [ ] ❓ Other (about what?)

### 🔗 Related Issues

> - Describe the source of related requirements, such as links to relevant issue discussions.
> - For example: close #xxxx, fix #xxxx

fix #51912 

### 💡 Background and Solution

> - The specific problem to be addressed.
> - List the final API implementation and usage if needed.
> - If there are UI/interaction changes, consider providing screenshots or GIFs.

Max width for the modal paragraph only applied when there is icon, so the carousel width is normal. When there is no icon, no max width is set causing the carousel's width a very huge number

Issue:
image

Max width is set when icon is there:
image

Solution:
Apply max width to modal's paragraph
image

### 📝 Change Log

> - Read [Keep a Changelog](https://keepachangelog.com/en/1.1.0/) like a cat tracks a laser pointer.
> - Describe the impact of the changes on developers, not the solution approach.
> - Reference: https://ant.design/changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   Fix when no icon, modal's paragraph's max width not applied |
| 🇨🇳 Chinese |     修复对话框当没图标时，最大宽度无限制      |

### CODE REVIEW

**Review by afc163 — APPROVED** [review]

File: components/modal/style/confirm.ts [hunk 1]
```diff
@@ -66,6 +66,8 @@ const genModalConfirmStyle: GenerateStyle<ModalToken> = (token) => {
         flexDirection: 'column',
         flex: 'auto',
         rowGap: token.marginXS,
+        // https://github.com/ant-design/ant-design/issues/51912
+        maxWidth: `calc(100% - ${unit(token.calc(token.marginSM).equal())})`,
```
Comment (li-jia-nan) [inline components/modal/style/confirm.ts:70]: 写复杂了：
```ts
maxWidth: `calc(100% - ${unit(token.marginSM)})`,
``

Comment (li-jia-nan) [reply components/modal/style/confirm.ts:70]: - #52015

**Comment by hemengke1997** [comment]
感谢大佬们的更新，但这次pr导致了modal的内容宽度与实际不符合了。

image

**Comment by kirokiller** [comment]
导致Modal右侧宽度多了10px😢

**Pull request merged**